src/common/components/elements/CurrentListings.tsx

```diff
@@ -0,0 +1,290 @@
+  { value: 'RECENTLY_LISTED', label: 'Recent listings' },
+];
+
+export type DiscoveryToolAction =
+  | {
+      type: 'INITIALIZE_LISTINGS';
+      payload: Listing[];
+    }
+  | {
+      type: 'LOAD_MORE_LISTINGS';
+    }
+  | {
+      type: 'FILTER';
+      payload: (keyof typeof filterFns)[];
+    }
+  | {
+      type: 'SORT';
+      payload: keyof typeof sortByFns;
+    };
+
+const initialState = (options?: {
+  filters?: FilterAction[];
+  sortBy?: SortByAction;
+}): DiscoveryToolState => {
+  const listingShells = Array(8)
+    .fill(null)
+    .map((_, i) => generateListingShell(i.toString()));
+  return {
+    allListings: listingShells,
+    filteredListings: listingShells,
+    listingsOnDisplay: listingShells,
+    cursor: 0,
+    // Array(8)
+    //   .fill(null)
+    //   .map((_, i) => generateListingShell(i.toString())),
+    filters: options?.filters || [],
+    sortBy: options?.sortBy || 'RECENTLY_LISTED',
+  };
+};
+
+function reducer(state: DiscoveryToolState, action: DiscoveryToolAction): DiscoveryToolState {
+  console.log(action.type, { prevState: state });
+  switch (action.type) {
+    case 'INITIALIZE_LISTINGS':
+      return {
+        ...state,
+        allListings: action.payload,
+        filteredListings: action.payload.sort(sortByFns[state.sortBy]),
+        listingsOnDisplay: [],
+      };
+    case 'LOAD_MORE_LISTINGS':
+      const newCursor = state.cursor + pageSize;
+      console.log('loading from', state.cursor, 'to', newCursor);
+      return {
+        ...state,
+        cursor: newCursor, // nr of items
+        listingsOnDisplay: [
+          ...state.listingsOnDisplay,
+          ...state.filteredListings.slice(state.cursor, newCursor),
+        ],
+      };
+    case 'FILTER':
+      const filteredListings = state.allListings
+        .filter((listing) => action.payload.every((filter) => filterFns[filter](listing)))
+        .sort(sortByFns[state.sortBy]);
+
+      return {
+        ...state,
+        cursor: 0,
+        filters: [...new Set(state.filters.concat(action.payload))], // de-duped filters
+        filteredListings: filteredListings,
+        listingsOnDisplay: filteredListings.slice(0, pageSize),
+      };
+    case 'SORT':
+      const sortedListings = state.filteredListings.sort(sortByFns[action.payload]);
+      return {
+        ...state,
+        cursor: 0,
+        sortBy: action.payload,
+        filteredListings: sortedListings,
+        listingsOnDisplay: sortedListings.slice(0, pageSize),
+      };
+    default:
+      throw new Error('Not a valid action for state');
+  }
+}
+
+export function CurrentListings() {
+  const router = useRouter();
+  const [loading, setLoading] = useState(false);
+
+  //Example query ?search=hello&filters=active_auctions,&sort_by=ending_soonest
+  const queryFilters: string[] =
+    (router.query['filters[]'] as string[]) || (router.query.filters as string)?.split(',') || [];
+  const querySortBy = (router.query.sort as string) || 'recent_listings';
```

Comment:
is `recent_listings` the right value here?

Reply:
Good catch, meant to be `recently_listed`